### PR TITLE
boosted cleaning fix for 75x

### PR DESCRIPTION
**Pull request by dsrankin**
Fixing const_cast issue.
Please approve as soon as possible, needed for data taking.
Backport of PR #10012 
@rappoccio @isando3

### CODE REVIEW

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
@drankincms 
also need a 74X PR!!

File: HLTrigger/JetMET/src/HLTJetCollectionsForBoostedLeptonPlusJets.cc [hunk 1]
```diff
@@ -94,119 +89,121 @@ HLTJetCollectionsForBoostedLeptonPlusJets<jetType>::produce(edm::Event& iEvent,
...
-           double deltaRPFConste = ClusP.DeltaR(PFJetConstP);  
-           if(deltaRPFConste < 0.001 && w==0){
-		   const_cast<LorentzVector&>(theJetCollection[j].p4()) = theJetCollection[j].p4() - clusCands[candNr]->p4();
-            w ++;
-           } //if
+  auto_ptr < JetCollection > allSelections(new JetCollection);
+
+  std::vector<size_t> usedCands;
+
+  std::cout<<"----------------NEW EVENT----------------\n";
```
Comment (Dr15Jones) [inline HLTrigger/JetMET/src/HLTJetCollectionsForBoostedLeptonPlusJets.cc]: Please remove cout statement

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
+1

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
@smuzaffar 
Could you please have a look at the jenkins tests of this PR - why they seem stuck?
Thanks!

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**